osquery/dispatcher/tests/task_schedule_tests.cpp

```diff
@@ -0,0 +1,145 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+#include <gtest/gtest.h>
+
+#include <osquery/config.h>
+#include <osquery/logger.h>
+#include <osquery/system.h>
+
+#include "osquery/dispatcher/task_schedule.h"
+
+namespace osquery {
+
+DECLARE_bool(disable_logging);
+
+class TaskScheduleTests : public testing::Test {
+  void SetUp() override {
+    logging_ = FLAGS_disable_logging;
+    FLAGS_disable_logging = true;
+  }
+
+  void TearDown() override {
+    FLAGS_disable_logging = logging_;
+  }
+
+ private:
+  bool logging_{false};
+};
+
+TEST_F(TaskScheduleTests, run_for_the_one_time) {
+  auto tasks = TaskSchedule{};
+  auto counter = int{0};
+  tasks.add(
+      [&counter](const auto scheduled_time) {
+        ++counter;
+        return 0;
+      },
+      0);
+  while (!tasks.isEmpty()) {
```

Comment:
I would make this much simpler. You actually don't need to sleep or have any loop. Just ensure

1. `tasks.isEmpty()` is `true` before `tasks.add()` and after `tasks.runNextNow()`.
2. `tasks.nextTimeToRun()` returns `0`.
3. `counter` is `1` as you're doing now.